lib/framework/SettingsService.h

```diff
@@ -1,58 +1,34 @@
 #ifndef SettingsService_h
 #define SettingsService_h
 
-#include <functional>
+#include <Arduino.h>
 
+#include <list>
+#include <functional>
 #ifdef ESP32
-#include <WiFi.h>
-#include <AsyncTCP.h>
-#elif defined(ESP8266)
-#include <ESP8266WiFi.h>
-#include <ESPAsyncTCP.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/semphr.h>
 #endif
 
-#include <ArduinoJson.h>
-#include <AsyncJson.h>
-#include <AsyncJsonCallbackResponse.h>
-#include <AsyncJsonWebHandler.h>
-#include <ESPAsyncWebServer.h>
-#include <SecurityManager.h>
-#include <SettingsPersistence.h>
-
 typedef size_t update_handler_id_t;
-typedef std::function<void(void)> SettingsUpdateCallback;
-static update_handler_id_t currentUpdateHandlerId;
+typedef std::function<void(String originId)> SettingsUpdateCallback;
+static update_handler_id_t currentUpdatedHandlerId;
 
 typedef struct SettingsUpdateHandlerInfo {
   update_handler_id_t _id;
   SettingsUpdateCallback _cb;
   bool _allowRemove;
   SettingsUpdateHandlerInfo(SettingsUpdateCallback cb, bool allowRemove) :
-      _id(++currentUpdateHandlerId),
-      _cb(cb),
-      _allowRemove(allowRemove){};
+      _id(++currentUpdatedHandlerId), _cb(cb), _allowRemove(allowRemove){};
 } SettingsUpdateHandlerInfo_t;
 
-/*
- * Abstraction of a service which stores it's settings as JSON in a file system.
- */
 template <class T>
-class SettingsService : public SettingsPersistence {
+class SettingsService {
  public:
```

Comment:
The struct T that holds settings could be quite complicated. We need a constructor that passes arguments from SettingsService to "T _settings". In my project for example it is a list of effects that current device supports.

```suggestion
 public:
   template<typename... Args>
   SettingsService::SettingsService(Args&&... args) : _settings(std::forward<Args>(args)...) {}
```